### PR TITLE
[TH2-5268] Added `fetchSize` option for query configuration.

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
+ added `fetchSize` option into query configuration.
+ added `defaultQueryFetchSize` option into root configuration.
+ updated:
  + th2 gradle plugin: `0.1.6` based on bom: `4.9.0`
  + common: `5.14.0-dev`
  + common-utils: `2.3.0-dev`
  + commons-dbcp2: `2.13.0`
+ updated jdbc:
  + postgresql: `42.7.4`
  + mysql-connector-j: `8.4.0`
  + ojdbc11: `23.6.0.24.10`
  + mssql-jdbc: `12.8.1.jre11`
+ migrated:
  + io.github.oshai:kotlin-logging: `5.1.4`

### CODE REVIEW

**Comment by lumber1000** [comment]
The local build fails for me without adding the following code to `build.gradle.kts` for the `grpc` subproject:
```kotlin
java {
    sourceCompatibility = JavaVersion.VERSION_11
    targetCompatibility = JavaVersion.VERSION_11
}
```

**Comment by Nikita-Smirnov-Exactpro** [comment]
> The local build fails for me without adding the following code to `build.gradle.kts` for the `grpc` subproject:
> 
> ```kotlin
> java {
>     sourceCompatibility = JavaVersion.VERSION_11
>     targetCompatibility = JavaVersion.VERSION_11
> }
> ```
> 
> Otherwise I get error: `Dependency resolution is looking for a library compatible with JVM runtime version 11, but 'project :grpc-read-db' is only compatible with JVM runtime version 17 or newer.`

What java version do you use locally for build project?

**Comment by lumber1000** [comment]
For some reason, JDK version 17 is being used for projects where no version is specified. I think it would be a good idea to explicitly specify the required version in build.gradle to avoid such issues.

**Comment by Nikita-Smirnov-Exactpro** [comment]
> ```kotlin
> sourceCompatibility = JavaVersion.VERSION_11
>     targetCompatibility = JavaVersion.VERSION_11
> ```

Added instructions

**Review by lumber1000 — APPROVED** [review]

File: core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/BaseQueryProvider.kt [hunk 1]
```diff
@@ -21,14 +21,15 @@ import com.exactpro.th2.read.db.core.QueryConfiguration
...
+import io.github.oshai.kotlinlogging.KotlinLogging
 import org.apache.commons.text.StringSubstitutor
 import org.apache.commons.text.lookup.StringLookup
 import java.sql.JDBCType
 import java.sql.SQLType
 
 class BaseQueryProvider(
-    queriesConfiguration: Map<QueryId, QueryConfiguration>
+    queriesConfiguration: Map<QueryId, QueryConfiguration>,
+    queryFetchSize: Int,
```
Comment (OptimumCode) [inline core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/BaseQueryProvider.kt:32]: suggestion: should be named `defaultQuertFetchSize` I think to make it more clear what it is for. Same for the parameter in the configuration class

**Pull request merged**